### PR TITLE
FlatGeobuf: Serialization cleanup

### PR DESCRIPTION
**Pull request by bjornharrtell**
~~Intends to fix issue 19198 in oss-fuzz.~~ (had proper fix in master already)

Also includes minor serialization cleanup that reduces file size somewhat and prepares for mixed geometry collections.

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp [hunk 1]
```diff
@@ -289,7 +288,9 @@ OGRGeometryCollection *GeometryReader::readGeometryCollection()
     if (parts == nullptr)
         return CPLErrorInvalidPointer("parts data");
     auto gc = new OGRGeometryCollection();
-    for (uoffset_t i = 0; i < parts->Length(); i++) {
+    if (parts == nullptr)
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: this test has been done just above

Comment (mloskot) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: Perhaps it was meant to check the gc. By the way, the gc could be captured with unique_ptr for exception safety

Comment (rouault) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: > Perhaps it was meant to check the gc

as gc results from a new, the only failure mode would be a C++ exception, not nullptr, right ?

Comment (mloskot) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: Yes, correct, but there is chance author's intention was to check that anyway.

Comment (bjornharrtell) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: Ah no it's a mistake when I merged locally.

File: gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp [hunk 2]
```diff
@@ -409,9 +408,9 @@ OGRPolyhedralSurface *GeometryReader::readPolyhedralSurface()
 
 OGRTriangulatedSurface *GeometryReader::readTIN()
 {
-    const auto pEnds = m_geometry->ends();
+    const auto ends = m_geometry->ends();
     const auto ts = new OGRTriangulatedSurface();
```
Comment (mloskot) [inline gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: By the way, why not to follow this exception safety convention that is already in use here?

https://github.com/OSGeo/gdal/blob/1a539bb944ebaa10d11bd32a8c78113c5dc3b9b4/gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufdataset.cpp#L185-L186

Comment (rouault) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp]: Merging as in current state. I'll let @bjornharrtell issue a potential follow-up PR if he feels like using more unique_ptr goodness

**Pull request merged**